agent/agent.go

```diff
@@ -18,6 +18,12 @@ const (
 	nodeUpdatePeriod             = 20 * time.Second
 )
 
+// Secrets is a map that keeps all the currenty available secrets to the agent
+type Secrets struct {
```

Comment:
Added a `secrets.go` under the agent package that contains the secret structure, along with some additional APIs that manipulate the map.